### PR TITLE
Some Stuff

### PR DESCRIPTION
**Pull request by OcelotWalrus**
This PR do what's been discussed in issue #18 and fixes aliases and optimize some stuff.

### CODE REVIEW

**Comment by Monarchz3ro** [comment]
change strings so that they're "groups" instead of "__groups" (same with the registry.json file)

File: kernel/master.py [hunk 1]
```diff
@@ -14,44 +14,58 @@
...
+        self.current_directory = self.__current_directory
+
+
     def boot_up(self):
         if self.checkxistence(self.boot):
             with open(self.boot, "r") as file:
                 temp = file.read()
                 if temp != syscheck.pathos_boot_module:
-                    self.cout("///FATAL_CRASH.KERNEL.PANIC///\nBoot file is corrupted. Please reinstall the system.")
+                    self.cout("///FATAL_CRASH.__kernel.PANIC///\nBoot file is corrupted. Please reinstall the system.")
```
Comment (OcelotWalrus) [inline kernel/master.py:68]:
```suggestion
                    self.cout("///FATAL_CRASH.KERNEL.PANIC///\nBoot file is corrupted. Please reinstall the system.")
```

File: kernel/master.py [hunk 2]
```diff
@@ -62,20 +76,20 @@ def create_default_environment_variables(self):
         self.shell_variables["$PATH"] = self.env_path_var
         self.shell_variables["$DATE"] = self.current_date
         self.shell_variables["$PRETTY_DATE"] = self.current_formatted_data
-        self.shell_variables["$REGISTRY"] = self.registry
+        self.shell_variables["$__registry"] = self.__registry
```
Comment (OcelotWalrus) [inline kernel/master.py:79]:
```suggestion
        self.shell_variables["$REGISTRY"] = self.__registry
```

File: kernel/master.py [hunk 3]
```diff
@@ -333,10 +349,10 @@ def __pathos_bus_locate_user_in_group(self, user):
         'locate the user in the group.'
         if user == "root":
             return "root"
-        with open(self.registry, "r") as file:
+        with open(self.__registry, "r") as file:
             reg_object = json.load(file)
-        for group in reg_object["groups"]:
-            if user in reg_object["groups"][group]:
+        for group in reg_object["__groups"]:
```
Comment (OcelotWalrus) [inline kernel/master.py:354]:
```suggestion
        for group in reg_object["groups"]:
```

File: kernel/master.py [hunk 4]
```diff
@@ -333,10 +349,10 @@ def __pathos_bus_locate_user_in_group(self, user):
         'locate the user in the group.'
         if user == "root":
             return "root"
-        with open(self.registry, "r") as file:
+        with open(self.__registry, "r") as file:
             reg_object = json.load(file)
-        for group in reg_object["groups"]:
-            if user in reg_object["groups"][group]:
+        for group in reg_object["__groups"]:
+            if user in reg_object["__groups"][group]:
```
Comment (OcelotWalrus) [inline kernel/master.py:355]:
```suggestion
            if user in reg_object["groups"][group]:
```

File: kernel/master.py [hunk 5]
```diff
@@ -349,69 +365,69 @@ def __pass_authenticated(self, password):
...
-        with open(self.registry, "r") as file:
+        with open(self.__registry, "r") as file:
             reg_object = json.load(file)
-        return reg_object["groups"][group][user]["password"]
+        return reg_object["__groups"][group][user]["password"]
     
-    def get_registry(self):
-        'get the registry of the system.'
-        with open(self.registry, "r") as file:
+    def get___registry(self):
```
Comment (OcelotWalrus) [inline kernel/master.py:399]:
```suggestion
    def get_registry(self):
```

File: kernel/master.py [hunk 6]
```diff
@@ -349,69 +365,69 @@ def __pass_authenticated(self, password):
...
+        'get the __registry of the system.'
```
Comment (OcelotWalrus) [inline kernel/master.py:400]:
```suggestion
        'get the registry of the system.'
```

File: kernel/master.py [hunk 7]
```diff
@@ -349,69 +365,69 @@ def __pass_authenticated(self, password):
...
+        with open(self.__registry, "r") as file:
             return json.load(file)
         
     def get_users(self):
         'get the users of the system.'
-        return self.get_registry()["users"]
+        return self.get___registry()["users"]
```
Comment (OcelotWalrus) [inline kernel/master.py:406]:
```suggestion
        return self.get_registry()["users"]
```

File: kernel/master.py [hunk 8]
```diff
@@ -349,69 +365,69 @@ def __pass_authenticated(self, password):
...
         'check if a file or directory exists. accepts a relpath string.'
-        ret = os.path.exists(os.path.join(self.root_dir, path))
+        ret = os.path.exists(os.path.join(self.__root_dir, path))
         if not ret:
             self.cout(f"Path given does not exist.")
         return ret
     
     def legal(self, filepath=str):
-        """Clamp to root. accepts a relpath string. Check if the specified path is within the root_dir. the path doesn't
+        """Clamp to root. accepts a relpath string. Check if the specified path is within the __root_dir. the path doesn't
```
Comment (OcelotWalrus) [inline kernel/master.py:416]:
```suggestion
        """Clamp to root. accepts a relpath string. Check if the specified path is within the root_dir. the path doesn't
```

File: kernel/master.py [hunk 9]
```diff
@@ -349,69 +365,69 @@ def __pass_authenticated(self, password):
...
         need to point to an existing path, just check if it's valid."""
-        ret = os.path.commonpath([os.path.abspath(filepath), self.root_dir]).replace("\\","/") == self.root_dir
+        ret = os.path.commonpath([os.path.abspath(filepath), self.__root_dir]).replace("\\","/") == self.__root_dir
         if not ret:
             self.cout(f"///SECURITY ERROR///\n3: Virtualisation Breakthrough Suppressed.")
         return ret
 
     def validated(self, filepath=str):
-        """check if the specified exists AND is within the root_dir."""
+        """check if the specified exists AND is within the __root_dir."""
```
Comment (OcelotWalrus) [inline kernel/master.py:424]:
```suggestion
        """check if the specified exists AND is within the root_dir."""
```

File: kernel/master.py [hunk 10]
```diff
@@ -420,13 +436,13 @@ def allowed(self, filepath, action, user, groups):
...
-            groups = data[filepath]["group"]
+            __groups = data[filepath]["group"]
             permissions = data[filepath]["permissions"]
 
             #get the perms of the file
-            with open(self.registry, "r") as file:
+            with open(self.__registry, "r") as file:
                 reg_object = json.load(file)
-                grps = reg_object["groups"]
+                grps = reg_object["__groups"]
```
Comment (OcelotWalrus) [inline kernel/master.py:445]:
```suggestion
                grps = reg_object["groups"]
```

File: kernel/master.py [hunk 11]
```diff
@@ -446,7 +462,7 @@ def allowed(self, filepath, action, user, groups):
         return 0
         
     def sprint_through(self):
-        "standalone tool to build the ecosystem.json file completely, assuming it doesn't exist in the first place or is empty."
+        "standalone tool to build the cosystem.json file completely, assuming it doesn't exist in the first place or is empty."
```
Comment (OcelotWalrus) [inline kernel/master.py:465]:
```suggestion
        "standalone tool to build the ecosystem.json file completely, assuming it doesn't exist in the first place or is empty."
```

File: kernel/master.py [hunk 12]
```diff
@@ -555,23 +571,23 @@ def get_meta_entry(self, path_to_entry):
             meta = json.load(file)
         return meta[path_to_entry.replace("\\","/")]
         
-    def get_ecosystem_data(self):
+    def get___ecosystem_data(self):
```
Comment (OcelotWalrus) [inline kernel/master.py:574]:
```suggestion
    def get_ecosystem_data(self):
```

File: kernel/master.py [hunk 13]
```diff
@@ -555,23 +571,23 @@ def get_meta_entry(self, path_to_entry):
             meta = json.load(file)
         return meta[path_to_entry.replace("\\","/")]
         
-    def get_ecosystem_data(self):
+    def get___ecosystem_data(self):
         with open(self.filesystem, "r") as file:
             meta = json.load(file)
             
-        # check for every items in the ecosystem
+        # check for every items in the __ecosystem
```
Comment (OcelotWalrus) [inline kernel/master.py:578]:
```suggestion
        # check for every items in the ecosystem
```

File: kernel/master.py [hunk 14]
```diff
@@ -555,23 +571,23 @@ def get_meta_entry(self, path_to_entry):
...
         # if the user has read permission to it
         for i in list(meta):
-            if not self.allowed(os.path.join(self.kernel, i), "r", self.__user, self.groups):
+            if not self.allowed(os.path.join(self.__kernel, i), "r", self.__user, self.__groups):
                 meta.pop(i)
         return meta
 
-    ### end of kernel methods - system methods start here
+    ### end of __kernel methods - system methods start here
```
Comment (OcelotWalrus) [inline kernel/master.py:585]:
```suggestion
    ### end of kernel methods - system methods start here
```

File: kernel/master.py [hunk 15]
```diff
@@ -585,33 +601,33 @@ def create_new_file(self, path_to_file, data:dict=None, contents=""):
...
+        shutil.rmtree(os.path.join(self.__current_directory, path_to_directory))
         self.delete_meta_entry(path_to_directory)
     
     def general_move(self, path_to_file, new_path, flags=None):
         'moves anything. accepts two relpath strings.'
-        if self.validated(os.path.join(self.current_directory, path_to_file)):
-            relative_path_to_file = os.path.join(self.current_directory, path_to_file).split('kernel/', 1)[1]
-            relative_new_path = os.path.join(self.current_directory, new_path).split('kernel/', 1)[1]
+        if self.validated(os.path.join(self.__current_directory, path_to_file)):
+            relative_path_to_file = os.path.join(self.__current_directory, path_to_file).split('__kernel/', 1)[1]
```
Comment (OcelotWalrus) [inline kernel/master.py:620]:
```suggestion
            relative_path_to_file = os.path.join(self.__current_directory, path_to_file).split('kernel/', 1)[1]
```

File: kernel/master.py [hunk 16]
```diff
@@ -585,33 +601,33 @@ def create_new_file(self, path_to_file, data:dict=None, contents=""):
...
+            relative_new_path = os.path.join(self.__current_directory, new_path).split('__kernel/', 1)[1]
```
Comment (OcelotWalrus) [inline kernel/master.py:621]:
```suggestion
            relative_new_path = os.path.join(self.__current_directory, new_path).split('kernel/', 1)[1]
```

File: kernel/master.py [hunk 17]
```diff
@@ -624,22 +640,22 @@ def general_copy(self, path_to_file, new_path):
     
     def get_file_contents(self, path_to_file):
         'get the contents of a file. accepts a relpath string.'
-        with open(os.path.join(self.current_directory, path_to_file).replace("\\","/"), "r") as file:
+        with open(os.path.join(self.__current_directory, path_to_file).replace("\\","/"), "r") as file:
             return file.read()
     
-    ## methods that expose the kernel to the bus
+    ## methods that expose the __kernel to the bus
```
Comment (OcelotWalrus) [inline kernel/master.py:646]:
```suggestion
    ## methods that expose the kernel to the bus
```

File: kernel/master.py [hunk 18]
```diff
@@ -656,22 +672,22 @@ def __pathos_bus_listdir_long(self, path):
...
+        os.mkdir(os.path.join(self.__current_directory, path))
     
     def __pathos_bus_mkdir_p(self, path):
         'pathos bus method to expose the mkdir -p command to the bus.'
-        os.makedirs(os.path.join(self.current_directory, path))
+        os.makedirs(os.path.join(self.__current_directory, path))
 
     def __pathos_bus_update_meta(self, path_to_entry, attribute, new_value):
-        'pathos bus method to expose the ecosystem to the bus.'
+        'pathos bus method to expose the __ecosystem to the bus.'
```
Comment (OcelotWalrus) [inline kernel/master.py:682]:
```suggestion
        'pathos bus method to expose the ecosystem to the bus.'
```

File: kernel/master.py [hunk 19]
```diff
@@ -656,22 +672,22 @@ def __pathos_bus_listdir_long(self, path):
...
         with open(self.filesystem, "r") as file:
             meta = json.load(file)
-        meta[os.path.relpath(path_to_entry,self.kernel).replace("\\","/")][attribute] = new_value
+        meta[os.path.relpath(path_to_entry,self.__kernel).replace("\\","/")][attribute] = new_value
         with open(self.filesystem, "w") as file:
             json.dump(meta, file, indent=4)
     
     def __pathos_bus_add_meta(self, path_to_entry, permissions, owner, group, size,last_modified):
-        'pathos bus method to expose the ecosystem to the bus.'
+        'pathos bus method to expose the __ecosystem to the bus.'
```
Comment (OcelotWalrus) [inline kernel/master.py:690]:
```suggestion
        'pathos bus method to expose the ecosystem to the bus.'
```

File: kernel/master.py [hunk 20]
```diff
@@ -714,27 +730,27 @@ def __pathos_bus_remove(self, path, r=False):
                     json.dump(memory_dump, file, indent=4)
 
 
-    ## os methods - methods that serve the exposed kernel via the bus
+    ## os methods - methods that serve the exposed __kernel via the bus
```
Comment (OcelotWalrus) [inline kernel/master.py:733]:
```suggestion
    ## os methods - methods that serve the exposed kernel via the bus
```

File: kernel/registry.json [hunk 21]
```diff
@@ -1,5 +1,5 @@
 {
-    "groups": {
+    "__groups": {
```
Comment (OcelotWalrus) [inline kernel/registry.json:2]:
```suggestion
    "groups": {
```

File: kernel/root/bin/locate.py [hunk 22]
```diff
@@ -8,8 +8,8 @@ def main(self:object, args: list[str]):
         return
     occurrence_string = args[0]
     list_of_files = []
-    ecosystem = self.get_ecosystem_data()
-    for item in list(ecosystem):
+    __ecosystem = self.get___ecosystem_data()
```
Comment (OcelotWalrus) [inline kernel/root/bin/locate.py:11]:
```suggestion
    ecosystem = self.get_ecosystem_data()
```

File: kernel/root/bin/locate.py [hunk 23]
```diff
@@ -8,8 +8,8 @@ def main(self:object, args: list[str]):
...
+    for item in list(__ecosystem):
```
Comment (OcelotWalrus) [inline kernel/root/bin/locate.py:12]:
```suggestion
    for item in list(ecosystem):
```

**Review by OcelotWalrus — COMMENTED** [review]
You want me to do this if I'm right? Sorry that I refactored without really checking what that was doing.

**Comment by Monarchz3ro** [comment]
aye, no worries, you're fine

**Comment by OcelotWalrus** [comment]
> aye, no worries, you're fine

?

**Comment by Monarchz3ro** [comment]
I meant to say, "Don't worry, you've corrected the mistakes, so you're fine"

**Comment by OcelotWalrus** [comment]
> I meant to say, "Don't worry, you've corrected the mistakes, so you're fine"

? I didn't. I've just proposed the changes. You're new to github.

**Pull request merged**